### ISSUE TITLE
type safety if/else not always working for me

### ISSUE THREAD

**Issue by nicferrier** [vultix/ts-results#5]
Firstly - this is great! I love it!

But I'm having an issue when I don't think I should, perhaps it's a newbie error in which case I apologize but:

```
    const certResponse:Result<PFXCert,Error> = await getCert(endpoint, dnsName);
    if (certResponse.err) {
        console.log("failed to get certificate");
        return;
    }
    else {
        const tlsOpts:PFXCert = certResponse.val;
        const {
            pfx: Buffer,
            passphrase: string
        } = tlsOpts;
        const secureOpts:SecureContextOptions = {
            pfx: tlsOpts.pfx,
            passphrase: tlsOpts.passphrase
        };
    }
```

gives me this compile error:

secret-recevier.ts(176,15): error TS2322: Type 'Error | PFXCert' is not assignable to type 'PFXCert'

But I thought the err check in the if clause meant that the else would ensure that TS knows val is bound to the OK?

**Comment by Grummfy** [comment]
same here, I finnaly do the !certRsponse.ok instead of certRsponse.err

**Comment by vultix** [comment]
This should be fixed in the latest `ts-results 3.0.0` release.

**Comment by arjunnaha** [comment]
I still seem to get this error. `!user.ok` works though...

(I'm on ^3.0.0)
